File: CS/CustomFunctionCriteriaOperator.Module/BusinessObjects/Employee.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

using DevExpress.ExpressApp.Security.Strategy;

namespace CustomFunctionCriteriaOperator.Module.BusinessObjects {
   [DefaultClassOptions(), DefaultProperty("UserName")]
   public class Employee : SecuritySystemUser {
      private Company company;

      public Employee(Session session)
         : base(session) {}

      [Association("Company-Employees", typeof(Company))]
      public Company Company {
         get {
            return company;
         }
         set {
            SetPropertyValue("Company", ref company, value);
         }
      }
   }

}

File: VB/CustomFunctionCriteriaOperator.Module/BusinessObjects/Employee.vb
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Imports DevExpress.ExpressApp.Security.Strategy

Namespace CustomFunctionCriteriaOperator.Module.BusinessObjects
   <DefaultClassOptions(), DefaultProperty("UserName")> _
   Public Class Employee
       Inherits SecuritySystemUser


      Private company_Renamed As Company

      Public Sub New(ByVal session As Session)
          MyBase.New(session)
      End Sub

      <Association("Company-Employees", GetType(Company))> _
      Public Property Company() As Company
         Get
            Return company_Renamed
         End Get
         Set(ByVal value As Company)
            SetPropertyValue("Company", company_Renamed, value)
         End Set
      End Property
   End Class

End Namespace
